### PR TITLE
Data processing

### PR DESCRIPTION
**Pull request by NduatiKagiri**
In this branch I:

- Included the associations in the models
- Added the following additional methods for the models.
    -  Users
        - A method that returns the 3 most recent posts for a given user.
    -  Posts
        - A method that updates the posts counter for a user.
        - A method which returns the 5 most recent comments for a given post.
    -  Comments
        - A method that updates the comments counter for a post.
    -  Likes
        - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (mikethreels) [inline app/models/user.rb]: - [ ] the foreign keys should be set in these lines of code.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,14 @@
 class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def user_counter_updater
+    total_posts = User.find_by(User_id).posts_counter
+    User.find_by(User_id).update(posts_counter: total_posts + 1)
```
Comment (mikethreels) [inline app/models/post.rb]: - [ ] [OPTIONAL] To make this a bit easier you might want to use the below code `user.increment!(:posts_counter)`

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi John, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb [hunk 3]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :user
```
Comment (BenMukebo) [inline app/models/post.rb:2]: - [OPTiONAL] The associations between models must be set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Which means 🤔 The Post belongs-to user through the `'author_id'` column of the Post table. Now, we need to show that, the  `'author_id'` reference  User class modal.
Your code should look like this:

```ruby
  # app/models/post.rb
  belongs_to :author, class_name: 'User'
```

File: app/models/like.rb [hunk 4]
```diff
@@ -1,2 +1,8 @@
 class Like < ApplicationRecord
+  belongs_to :user
```
Comment (BenMukebo) [inline app/models/like.rb:2]: - [OPTONAL] The same here 🤔 Likes belongs-to user through the `'author_id'` column of the Like table. Now, we need to show that, the  `'author_id'` reference  User class modal.
Your code should look like this 👇🏻 :

```ruby
 # app/models/like.rb
  belongs_to :author, class_name: 'User'
```

File: app/models/comment.rb [hunk 5]
```diff
@@ -1,2 +1,8 @@
 class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (BenMukebo) [inline app/models/comment.rb:2]: - [OPTONAL] The same here 🤔  The Comments belongs-to user through the `'author_id'` column of the Comment table. Now, we need to show that, the  `'author_id'` reference  User class modal.
Your code should look like this 👇🏻 :

```ruby
  # app/models/comment.rb
  belongs_to :author, class_name: 'User'
```

**Review by BenMukebo — APPROVED** [review]
### Hi @NduatiKagiri, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- The associations between models is set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).:heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL]  Please kindly update your `readme file` to reflect the changes for this milestone as this will make your `readme file` more informative and professional. Check out this [template](https://github.com/microverseinc/readme-template) and adapt it to your project. Replace all the information in the various sections of the template with information pertaining to your project. 

- The README file It is the first file a person will see when they encounter your project, so it should be fairly brief but detailed. It will make your project stand out from a bunch of others.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**